src/test/anovos/feature_recommender/test_feature_exploration.py

```diff
@@ -0,0 +1,177 @@
+import pandas as pd
+import pytest
+from anovos.feature_recommender.feature_exploration import (
+    list_all_industry,
+    list_all_usecase,
+    list_all_pair,
+    process_usecase,
+    process_industry,
+    list_usecase_by_industry,
+    list_industry_by_usecase,
+    list_feature_by_industry,
+    list_feature_by_usecase,
+    list_feature_by_pair,
+)
+from anovos.feature_recommender.featrec_init import model_download
+
+
+model_download()
+
+
+def test_list_all_industry():
+    test_df = list_all_industry()
+    assert len(test_df) > 0
+    assert "Industry" in test_df.columns
+    assert test_df.iloc[:, 0].nunique() == len(test_df)
+    for i in range(len(test_df)):
+        assert "," not in test_df.iloc[i, 0]
+
+
+def test_list_all_usecase():
+    test_df = list_all_usecase()
+    assert len(test_df) > 0
+    assert "Usecase" in test_df.columns
+    assert test_df.iloc[:, 0].nunique() == len(test_df)
+    for i in range(len(test_df)):
+        assert "," not in test_df.iloc[i, 0]
+
+
+def test_list_all_pair():
+    test_df = list_all_pair()
+    assert len(test_df) > 0
+    assert "Industry" in test_df.columns
+    assert "Usecase" in test_df.columns
+    assert test_df.groupby(["Industry", "Usecase"]).ngroups == len(test_df)
+
+
+def test_process_usecase():
+    assert process_usecase("fraud", semantic=True) == "fraud detection"
+    assert process_usecase("fraud", semantic=False) == "fraud"
```

Comment:
Can we also have some negative checks. like assert not process_usecase("fraud", semantic=False) == <something_unrelated>